barretenberg/cpp/src/barretenberg/sumcheck/sumcheck_round.hpp

```diff
@@ -171,19 +193,21 @@ template <typename Flavor> class SumcheckProverRound {
             Utils::zero_univariates(thread_univariate_accumulators[thread_idx]);
             // Construct extended univariates containers; one per thread
             ExtendedEdges extended_edges;
-            size_t start = thread_idx * iterations_per_thread;
-            size_t end = (thread_idx + 1) * iterations_per_thread;
-            for (size_t edge_idx = start; edge_idx < end; edge_idx += 2) {
-                extend_edges(extended_edges, polynomials, edge_idx);
-                // Compute the \f$ \ell \f$-th edge's univariate contribution,
-                // scale it by the corresponding \f$ pow_{\beta} \f$ contribution and add it to the accumulators for \f$
-                // \tilde{S}^i(X_i) \f$. If \f$ \ell \f$'s binary representation is given by \f$ (\ell_{i+1},\ldots,
-                // \ell_{d-1})\f$, the \f$ pow_{\beta}\f$-contribution is \f$\beta_{i+1}^{\ell_{i+1}} \cdot \ldots \cdot
-                // \beta_{d-1}^{\ell_{d-1}}\f$.
-                accumulate_relation_univariates(thread_univariate_accumulators[thread_idx],
-                                                extended_edges,
-                                                relation_parameters,
-                                                gate_sparators[(edge_idx >> 1) * gate_sparators.periodicity]);
+            for (size_t chunk_idx = 0; chunk_idx < num_of_chunks; chunk_idx++) {
```

Comment:
It feels like it's not the first time I've seen this kind of chunking in BB. We should probably (eventually) track down the uses and write a variant of `parallel_for` that does it for you.